### PR TITLE
[ISSUE-125] Add allow_restricted_indices setting to security role

### PR DESCRIPTION
**Pull request by RobsonSutton**
- Add allow_restricted_indices setting (identified as missing: https://github.com/elastic/terraform-provider-elasticstack/issues/125)

### CODE REVIEW

**Comment by RobsonSutton** [comment]
Also, https://github.com/elastic/elasticsearch/pull/88751 raised to update the create / update security role docs to include allow_restricted_indices attribute.

@Kushmaro / @olksdr - Hi both 👋 Please can you review and provide some feedback on this one when you get a chance? Thanks!

File: internal/elasticsearch/security/role_test.go [hunk 1]
```diff
@@ -58,8 +60,9 @@ resource "elasticstack_elasticsearch_security_role" "test" {
   cluster = ["all"]
 
   indices {
-    names      = ["index1", "index2"]
-    privileges = ["all"]
+    names                    = ["index1", "index2"]
+    privileges               = ["all"]
+		allow_restricted_indices = true
```
Comment (tobio) [inline internal/elasticsearch/security/role_test.go]: nit: 

```suggestion
    allow_restricted_indices = true
```

**Review by tobio — COMMENTED** [review]
LGTM, minor style nit to be consistent with the existing code.

**Comment by tobio** [comment]
jenkins test this

File: internal/elasticsearch/security/role_test.go (hunk 1 above)
Comment (RobsonSutton) [reply internal/elasticsearch/security/role_test.go]: Completely missed this! Switched tabs to spaces to resolve 👍

**Comment by tobio** [comment]
jenkins test this

**Comment by tobio** [comment]
@RobsonSutton sorry for the back and forth, I'm new to this repo. Are you able to add a changelog entry for this PR as well please?

**Comment by RobsonSutton** [comment]
@tobio - No probs! apologies, completely missed this!

**Review by tobio — APPROVED** [review]

**Comment by tobio** [comment]
jenkins test this

**Pull request merged**